c++/src/capnp/compat/websocket-rpc.c++

```diff
@@ -0,0 +1,107 @@
+#include <capnp/compat/websocket-rpc.h>
+#include <kj/io.h>
+#include <capnp/serialize.h>
+
+namespace capnp {
+
+WebSocketMessageStream::WebSocketMessageStream(kj::WebSocket& socket, size_t maxMessageReceiveSize)
+  : socket(socket),
+    maxMessageReceiveSize(maxMessageReceiveSize)
+  {};
+
+kj::Promise<kj::Maybe<MessageReaderAndFds>> WebSocketMessageStream::tryReadMessage(
+    kj::ArrayPtr<kj::AutoCloseFd> fdSpace,
+    ReaderOptions options, kj::ArrayPtr<word> scratchSpace) {
+  return socket.receive(maxMessageReceiveSize)
+    .then([options, scratchSpace](auto msg) -> kj::Promise<kj::Maybe<MessageReaderAndFds>> {
+      KJ_SWITCH_ONEOF(msg) {
+        KJ_CASE_ONEOF(closeMsg, kj::WebSocket::Close) {
+          return kj::Maybe<MessageReaderAndFds>();
+        }
+        KJ_CASE_ONEOF(str, kj::String) {
+          KJ_FAIL_REQUIRE(
+            "Unexpected websocket text message; expected only binary messages."
+          );
+          break;
+        }
+        KJ_CASE_ONEOF(bytes, kj::Array<byte>) {
+          kj::ArrayPtr<byte> ptr(bytes);
+          if(reinterpret_cast<uintptr_t>(&bytes[0]) % alignof(word) == 0
```

Comment:
Nitpick: We put a space between a control keyword (`if`, `for`, ...) at a `(`:

```suggestion
          if (reinterpret_cast<uintptr_t>(&bytes[0]) % alignof(word) == 0
```

I don't have a good reason for this rule... it comes from Google style and I never unlearned it.

(Several other instances of this in the PR.)